orttraining/orttraining/test/python/orttraining_test_orttrainer_bert_toy_onnx.py

```diff
@@ -454,6 +454,18 @@ def testToyBertCheckpointLoadZero():
     actual_eval_loss = trainer.eval_step(input_ids, segment_ids, input_mask, masked_lm_labels, next_sentence_labels)
     actual_eval_loss = actual_eval_loss.cpu().numpy().item(0)
 
+    checkpoint_files = sorted(checkpoint._list_checkpoint_files(ckpt_dir, 'bert_toy_lamb'))
+    loaded_state_dict = checkpoint.experimental_state_dict(trainer)
+    loaded_state_dict = checkpoint._split_state_dict(loaded_state_dict)
+
+    for f in checkpoint_files:
+        rank_state_dict = torch.load(f, map_location=torch.device("cpu"))
+        rank_state_dict = checkpoint._split_state_dict(rank_state_dict)
+
+        for k,v in rank_state_dict['fp16_param'].items():
+            fp32_name = k.split('_fp16')[0]
+            assert_allclose(v, loaded_state_dict['fp32_param'][fp32_name])
```

Comment:
going from mixed precision to full precision distributed training is yielding same weights between fp16 pre and fp32 post? This does not allign with my findings in the trainer checkpoint migration matrix excel sheet.

Reply:
We can investigate the bugs you have found separately. I have added the script for generating the checkpoints as well in the PR, please feel free to regenerate and run the test, but it passes in the CI as well as locally

Reply:
After speaking to @ashbhandare and going through my test scenarios, it seems I might have worked with a corrupt checkpoint file which was causing my test to fail. Some of the error scenarios I reported are now no longer failing. Investigating more on this and will update the matrix as well as the bug I filed.